### PR TITLE
Feature/reactor 331

### PR DESCRIPTION
**Pull request by tiyberius**
### Description

- [ ] Existing GitHub Issue? Please provide #: 

### Acceptance checklist

- [ ] Feedback addressed
- [ ] Tests added or modified appropriately
- [ ] All tests passed

### CODE REVIEW

File: lambda_tools.py [hunk 1]
```diff
@@ -181,7 +181,7 @@ def aws_lambda_wrapper(event, context):
 
             client_details = get_client_details(event)
             logger.info(f"lambda_handler.{lambda_func.__name__} | {client_details}")
-            logger.debug(event)
+            logger.debug(f'Lambda event: {event}')
```
Comment (silvexis) [inline lambda_tools.py]: Really intense change

File: reactor/aws/tools.py [hunk 2]
```diff
@@ -76,22 +76,22 @@ def verify_sqs_access(aws, queue_name):
         return ResourceAccess.NOT_OK, "Could not access SQS Queue"
 
 
-def get_name_from_arn(arn):
+def get_object_name_from_arn(arn):
```
Comment (silvexis) [inline reactor/aws/tools.py]: I _almost_ did this...._almost_

File: reactor/aws/tools.py [hunk 3]
```diff
@@ -478,6 +478,14 @@ def get_lambda_functions(aws):
             for f in res['Functions']}
 
 
+def get_reactor_s3_bucket_name(namespace, account_id):
```
Comment (silvexis) [inline reactor/aws/tools.py]: ok this is a better approach, can you remove/replace the use of REACTOR_BUCKET_FORMAT constant from constants.py?

File: reactor/features/resources/curate.py [hunk 4]
```diff
@@ -124,12 +122,21 @@ def _process_new_resource_id(resource_id, timestamp, service_type, environment_i
...
         return handle_no_plugin(namespace, payload, resource_id, environment_id)
     except Exception as error:  # We don't know what the fuck happened
-        logger.error(f'curate._process_new_resource_id | Unknown error {error}')
+        logger.error(f'curate._process_new_resource_id | Unknown error {error} processing resource id {resource_id}')
         new_disposition = reactor.constants.RESOURCE_DISPOSITION_ERROR
         dbh.set_disposition(resource_id, new_disposition, timestamp)
         return "error"
 
 
+def write_resource_to_s3(namespace, resource_id, environment_id, account_id, resource):
```
Comment (silvexis) [inline reactor/features/resources/curate.py]: rename account_id to reactor_account_id

File: reactor/features/resources/curate.py [hunk 5]
```diff
@@ -18,32 +19,33 @@
...
+from reactor.aws.tools import get_service_type_from_arn, get_reactor_s3_bucket_name
 from reactor.database.models import dynamodb
 
 logger = lambda_tools.setup_logging('reactor')
 
 _ENRICH_FUNCTION_TEMPLATE = 'resources-enrich-plugin-{}'
 
 
-def curate(event, namespace):
+def curate(records, account_id, namespace):
```
Comment (silvexis) [inline reactor/features/resources/curate.py]: rename account_id to reactor_account_id

File: reactor/features/resources/plugins/base.py [hunk 6]
```diff
@@ -70,23 +69,23 @@ def reconcile(self, data, disposition=None):
...
-        if len(response['Items']) > 0:
-            logger.debug(f"reconcile | I'm not the latest ({len(response['Items'])} before me), "
+        if len(newer_resources['Items']) > 0:
+            logger.debug(f"reconcile | I'm not the latest ({len(newer_resources['Items'])} before me), "
                          f"exiting! {self.resource_id}:{self.timestamp}")
             return
 
         # get all versions of this resource
-        response = resources.table.query(
+        all_version = resources.table.query(
```
Comment (silvexis) [inline reactor/features/resources/plugins/base.py]: all_version**s** ?

File: lambda_tools.py (hunk 1 above)
Comment (tiyberius) [reply lambda_tools.py]: But seriously. I was looking at the logs and couldn't figure out where the hell that was coming from

File: reactor/aws/tools.py (hunk 2 above)
Comment (tiyberius) [reply reactor/aws/tools.py]: I'm in ur head

File: reactor/aws/tools.py (hunk 3 above)
Comment (tiyberius) [reply reactor/aws/tools.py]: done. will be fixed in next commit

File: reactor/features/resources/curate.py (hunk 5 above)
Comment (tiyberius) [reply reactor/features/resources/curate.py]: done

File: reactor/features/resources/curate.py (hunk 4 above)
Comment (tiyberius) [reply reactor/features/resources/curate.py]: done

File: reactor/features/resources/plugins/base.py (hunk 6 above)
Comment (tiyberius) [reply reactor/features/resources/plugins/base.py]: ALL THE VERSIONS!!!

All versions of that resource? What would you prefer

File: lambda_tools.py (hunk 1 above)
Comment (silvexis) [reply lambda_tools.py]: maybe then logger.debug(f'aws_lambda_wrapper | {event}')?

Comment (tiyberius) [reply lambda_tools.py]: alright I can do that

**Pull request merged**